server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesFetcher.java

```diff
@@ -37,9 +37,11 @@
  */
 class FieldCapabilitiesFetcher {
     private final IndicesService indicesService;
+    private final IndexFieldCapabilities.Deduplicator fieldDeduplicator;
 
-    FieldCapabilitiesFetcher(IndicesService indicesService) {
+    FieldCapabilitiesFetcher(IndicesService indicesService, boolean dedupFields) {
```

Comment:
why do we need to enable deduplication conditionally?

Reply:
I disabled the deduplication for index-level requests from old nodes (i.e., before 7.16) as the deduplication won't help. However, I think we should always enable the deduplication for simplicity. I pushed https://github.com/elastic/elasticsearch/pull/88238/commits/76a0bd169fb488c279482484111e761f57ff3074.